src/osd/PeeringState.cc

```diff
@@ -2406,13 +2413,15 @@ bool PeeringState::choose_acting(pg_shard_t &auth_log_shard_id,
 	get_osdmap());
     }
   }
-  while (want.size() > pool.info.size) {
-    // async recovery should have taken out as many osds as it can.
-    // if not, then always evict the last peer
-    // (will get synchronously recovered later)
-    psdout(10) << __func__ << " evicting osd." << want.back()
-               << " from oversized want " << want << dendl;
-    want.pop_back();
+  if (!pool.info.is_stretch_pool()) {
+    while (want.size() > pool.info.size) {
```

Comment:
I don't really understand this.  This is here because we choose to leave the extra want set osds in until async recovery makes its selections (22c8cda).  Is there something later on that handles this if we calc_replicated_acting_stretch includes extra osds?  Can calc_replicated_acting_stretch include extra osds?

Reply:
Ah, you are right! I misunderstood what was going on here when I made that change and should have looked more closely. I removed it and replaced it with a more verbose comment on the logic.